pkg/synapse/synapse.go

```diff
@@ -172,26 +173,36 @@ func (s *synapse) messageReader(normalCloser chan struct{}, conn *websocket.Conn
 		}
 		return nil
 	})
-
+	duplicateConnectionChan := make(chan struct{})
 	for {
-		_, msg, err := conn.ReadMessage()
-		if err != nil {
-			if websocket.IsCloseError(err, websocket.CloseNormalClosure) {
-				s.logger.Debugf("Normal closure occurred...........")
-				normalCloser <- struct{}{}
-				return
-			}
-			s.logger.Errorf("disconnecting from lambdatest server. error in reading message %v", err)
+		select {
+		case <-duplicateConnectionChan:
```

Comment:
how are we identifying duplicate connection?

Reply:
We are asserting over errMsg received from lambdatest server